_studio/shared/include/mfxvideo++int.h

```diff
@@ -499,6 +529,12 @@ class VideoUSER
     // Get the plugin's threading policy
     virtual
     mfxTaskThreadingPolicy GetThreadingPolicy(void) {return MFX_TASK_THREADING_DEFAULT;}
+    virtual
+    mfxStatus GetThreadNum(mfxU32& threadNum) {
+        threadNum = std::thread::hardware_concurrency();
```

Comment:
actually std::thread::hardware_concurrency could return 0, so I believe it should be
```
// some mediasdk components require 2 threads as minimum
threadNum = std::max(std::thread::hardware_concurrency(), 2u);
```

Why we returning status here? Maybe better to return actual number of threads?

Reply:
thank's for suggestion for std::max

and you saw the answer to the second question in some other patch: we may not always determine number of threads for the legacy components. for these cases we will return an error.

Reply:
yes, I got it